protocol/scripts/deploys/helpers.ts

```diff
@@ -269,6 +270,32 @@ export async function deployAndMine<
   );
   console.log('********************\n');
 
+  // If we are on hardhat chain
+  if (network.config.chainId === 4 && process.env.ETHERSCAN_API_KEY) {
+  // Wait for a few blocks to be mined before verifying
+  await contract.deployTransaction.wait(6)
+  // Verify the contract on Etherscan
+  const response = await axios.post(
+    `https://api.etherscan.io/api?module=contract&action=verifysourcecode&apikey=${process.env.ETHERSCAN_API_KEY}`,
+    {
+      contractaddress: contract.address,
+      sourceCode: (contract as any).deployedBytecode,
+      contractname: name,
+      compilerversion: 'v0.7.0+commit.9e61f92b',
+      optimizationUsed: 1,
+      runs: 200,
+      constructorArguements: renderedArgs,
```

Comment:
Typo in arguements

Also need to check this escapes multi word string args correctly

Reply:
And also for arrays